ruby/reschedule_or_cancel_unfilled_service.rb

```diff
@@ -0,0 +1,216 @@
+# Service for handling the automatic rescheduling/canceling of unfilled bookings.
+module RescheduleOrCancelUnfilledService
+  MAX_OFFSET = 8 * 7 # up to 8 weeks away
+
+  class << self
+    def auto_rbu_or_cbu_candidate?(booking, date_start: booking.date_start)
+      return false unless booking.confirmed
+      return false if booking.provider.present?
+
+      window = auto_rbu_or_cbu_minutes(region: booking.region, service: booking.service).minutes
+
+      ((date_start - window)...date_start).cover?(Time.current)
+    end
+
+    # Get bookings in need of auto-RBU/CBU
+    # @return array of eligible bookings
+    def bookings_to_reschedule_or_cancel
+      regions  = Region.select { |region| ConfigParam.regional_auto_rbu?(region) }
+      start_at = Time.current
+      window = [RegionAutoAcceptWindow.maximum(:auto_rbu_minutes).to_i, RegionAutoAcceptWindow::AUTO_RBU_WINDOW_MINUTES].max
+      end_at   = start_at + window.minutes
+
+      Booking.unfilled_by_regions_and_time(regions, start_at, end_at).
+        select{ |booking| booking.auto_rbu_or_cbu? } # This is not great, but is fast enough.
+    end
+
+    # Automatically RBU (Reschedule By Us) or CBU (Cancel By Us) an unfilled booking. Attempts to
+    # reschedule the booking to a suitable date. If that's not possible, cancels the booking.
+    #
+    # @param booking [Booking] unfilled booking to be RBU/CBU'd
+    def reschedule_or_cancel(booking)
+      return if MultipleProviderBookingService.try_double_assign_pro_to_prevent_dual_pro_rbu(booking)
+      return if !ConfigParam.regional_auto_rbu?(booking.region)
```

Comment:
I suggest two options here:
- Moving these conditions to separate method with a meaningful name with a boolean return value.
- We could also add a && for those two conditions.